src/client/Microsoft.Identity.Client/WwwAuthenticateParameters.cs

```diff
@@ -25,35 +25,15 @@ public class WwwAuthenticateParameters
         /// Resource for which to request scopes.
         /// This is the App ID URI of the API that returned the WWW-Authenticate header.
         /// </summary>
+        [Obsolete("The client apps should know which App ID URI it requests scopes for.", true)]
         public string Resource { get; set; }
 
         /// <summary>
         /// Scopes to request.
         /// If it's not provided by the web API, it's computed from the Resource.
         /// </summary>
-        public IEnumerable<string> Scopes
-        {
-            get
-            {
-                if (_scopes != null)
-                {
-                    return _scopes;
-                }
-
-                if (!string.IsNullOrEmpty(Resource))
-                {
-                    return new[] { $"{Resource}/.default" };
-                }
-
-                return null;
-            }
-            set
-            {
-                _scopes = value;
-            }
-        }
-
-        private IEnumerable<string> _scopes;
+        [Obsolete("The client apps should know which scopes to request for.", true)]
```

Comment:
Obsoleting with error - is this okay, in this case?

Reply:
Yes, I think so.